### DISCUSSION TITLE
delay in publishing messages when changing topics

### DISCUSSION THREAD

**Discussion by ghost** [emqx/emqx#4392] [Q&A]
use mqtt connection, the first message publish to a topic takes 6-7 secs to complete. subsequent message publish to the same topic is instantaneous. But if the topic is changes on the same connection, the first publishing takes 6-7 secs. This is using QOS=1. In QOS=0, the publish is instantaneous, but the subscribe takes the 6-7 second delay. Anyone has come across this issue?

**Comment by zmstone** [comment]
Hi @anantha-wr could you be more specific about your test scenario ?
e.g. broker server profile?
single node, or clustered? 
which client is it in your test?
publisher and consumer client properties?

Maybe you can enable debug level logging to see the timing of MQTT packets?

**Comment by ghost** [comment]
Thanks for reaching back to me.
We setup another emqx cluster to debug the issue.
What we discovered was a strange issue. If http-authentication is turned
on, the delay manifested. When http-authentication is turned off, it went
away.
We have switched to JWT authentication to get around this problem

<https://www.nuvizz.com>

*Anantha Rao*COO & CTO
(404) 937-1971 <4049371971+x4402> x4402
***@***.***
www.nuvizz.com

**Reply by zmstone** [reply]
Hi @anantha-wr 

Was there ACL configured ?
one explanation could be: the ACL is cached by the connection process,
so, it only has a hiccup for the first publish.

However, this does not explain why QoS 0 messages gets though quickly though.

I'm glad that you found a workaround.
We'd love to learn more details about this issue though,
 -- if you happen to give http-auth another try

**Comment by ghost** [comment]
ACL was not configured. The plugin was installed/ started.

<https://www.nuvizz.com>

*Anantha Rao*COO & CTO
(404) 937-1971 <4049371971+x4402> x4402
***@***.***
www.nuvizz.com
